newsroom/notifications/services.py

```diff
@@ -14,36 +16,38 @@ async def create_or_update(self, entries: list[dict[str, Any]]) -> list[str]:
         Save the given notifications entries into the database. If the notification entry already
         exists in the database then it proceed to execute an update instead
         """
-        from newsroom.users import UsersService
 
         now = utcnow()
         ids = []
+        users = await get_user_dict_async()
 
         for doc in entries:
-            user_id = doc["user"]
-            user = await UsersService().find_by_id(user_id)
-
+            user = users.get(ObjectId(doc["user"]))
             if user is None or not user.receive_app_notifications:
                 continue
 
-            notification_id = f"{user_id}_{doc['item']}"
+            notification_id = f"{user.id}_{doc['item']}"
             original = await self.find_by_id(notification_id)
 
             if original:
+                updates = {
+                    "_created": now,
+                    "action": doc.get("action") or original.action,
+                }
+                if "data" in doc:
+                    updates["data"] = doc["data"] or {}
                 await self.update(
                     notification_id,
-                    updates={
-                        "_created": now,
-                        "action": doc.get("action") or original.action,
-                        "data": doc.get("data") or original.data,
-                    },
+                    updates=updates,
```

Comment:
nit: I think the second param is the updates dict so to simplify and make the `self.update` one-liner
```suggestion
                    updates,
```